tikv: add new dynamic configure for flow-control (#18814)
pingcap/docs: dynamic-config.md

| `server.raft-msg-max-batch-size` | Sets the maximum number of Raft messages that are contained in a single gRPC message |
| `server.simplify-metrics`        | Controls whether to simplify the sampling monitoring metrics                   |
| `storage.block-cache.capacity` | The size of shared block cache (supported since v4.0.3) |
| storage.flow-control.enable | Determines whether to enable the flow control mechanism |

> | storage.flow-control.memtables-threshold | The threshold of kvDB memtable numbers that triggers flow control |
> | storage.flow-control.l0-files-threshold | The threshold of kvDB L0 file numbers that triggers flow control |

Reviewer: ```suggestion
| storage.flow-control.memtables-threshold | The maximum number of kvDB memtables that trigger flow control |
| storage.flow-control.l0-files-threshold | The maximum number of kvDB L0 files that trigger flow control |
```

> | storage.flow-control.soft-pending-compaction-bytes-limit | The threshold of kvDB pending compaction bytes that triggers flow control |

Reviewer: ```suggestion
| storage.flow-control.soft-pending-compaction-bytes-limit | The threshold of kvDB pending compaction bytes that triggers flow control mechanism to reject some write requests |
```

> | storage.flow-control.hard-pending-compaction-bytes-limit | The threshold of kvDB pending compaction bytes that triggers flow control mechanism to reject all write requests and report the `ServerIsBusy` error |

Reviewer: ```suggestion
| storage.flow-control.hard-pending-compaction-bytes-limit | The threshold of kvDB pending compaction bytes that triggers flow control mechanism to reject all write requests |
```

Reviewer: keep it consistent with the Chinese PR: https://github.com/pingcap/docs-cn/pull/18540/files

> | storage.flow-control.enable | Determines whether to enable the flow control mechanism. |
> | storage.flow-control.memtables-threshold | The threshold of kvDB memtable numbers that triggers flow control. |
> | storage.flow-control.l0-files-threshold | The threshold of kvDB L0 file numbers that triggers flow control. |
> | storage.flow-control.soft-pending-compaction-bytes-limit | The threshold of kvDB pending compaction bytes that triggers flow control. |
> | storage.flow-control.hard-pending-compaction-bytes-limit | The threshold of kvDB pending compaction bytes that triggers flow control mechanism to reject all write requests and reports the `ServerIsBusy` error. |

Reviewer 2: ```suggestion
| storage.flow-control.enable | Determines whether to enable the flow control mechanism |
| storage.flow-control.memtables-threshold | The threshold of kvDB memtable numbers that triggers flow control |
| storage.flow-control.l0-files-threshold | The threshold of kvDB L0 file numbers that triggers flow control |
| storage.flow-control.soft-pending-compaction-bytes-limit | The threshold of kvDB pending compaction bytes that triggers flow control |
| storage.flow-control.hard-pending-compaction-bytes-limit | The threshold of kvDB pending compaction bytes that triggers flow control mechanism to reject all write requests and report the `ServerIsBusy` error |
```

Keep the format consistent with other rows in this table.
